src/power_grid_model/validation/rules.py

```diff
@@ -754,6 +754,53 @@ def all_finite(data: SingleDataset, exceptions: dict[ComponentType, list[str]] |
     return errors
 
 
+def subset_of_fields_are_missing(data: SingleDataset, fields : list[str], component_type : ComponentType):
```

Comment:
probably better to use a name with a different default:

```suggestion
def no_strict_subset_missing(data: SingleDataset, fields : list[str], component_type : ComponentType):
```
or something like
```suggestion
def all_or_none_provided(data: SingleDataset, fields : list[str], component_type : ComponentType):
```

Reply:
check